pontoon/base/models.py

```diff
@@ -2588,7 +2591,22 @@ def map_entities(cls, locale, entities, visible_entities=None):
         for entity in entities:
             translation_array = []
 
-            if entity.string_plural == "":
+            if preferred_source_locale == "":
+                string = entity.string
+                string_plural = entity.string_plural
+            else:
+                preferredSourceLocale = Locale.objects.filter(
+                    code=preferred_source_locale).values_list('pk')[0][0]
+                string = Translation.objects.filter(locale=preferredSourceLocale, entity=entity)
+                if not string:
+                    string = entity.string
+                    string_plural = entity.string_plural
+                else:
+                    string = string.values_list('string')[0]
+                    # replace with Source Locale Translated Plural String
+                    string_plural = entity.string_plural
```

Comment:
You should remove this entire if-else block and only keep default string values:
```python
original = entity.string
original_plural = entity.string_plural
```